op-node/rollup/derive/calldata_source.go

```diff
@@ -61,19 +67,22 @@ type DataSource struct {
 // If there is an error, it will attempt to fetch the result on the next call to `Next`.
 func NewDataSource(ctx context.Context, log log.Logger, cfg *rollup.Config, fetcher L1TransactionFetcher, block eth.BlockID, batcherAddr common.Address) DataIter {
 	_, txs, err := fetcher.InfoAndTxsByHash(ctx, block.Hash)
+
+	dsCfg := DataSourceConfig{cfg.L1Signer(), cfg.BatchInboxAddress}
```

Comment:
This is probably a reasonable suggestion - we could take a `DataSourceConfig` as a param to `NewDataSource` and also use it instead of `rollup.Config` in `DataSourceFactory`.  Given where the factory is created currently, I'd probably keep `NewDataSourceFactory` taking a `rollup.Config` and have it create the `DataSourceConfig` though.